planner/core/rule_build_key_info.go

```diff
@@ -247,9 +264,12 @@ func (ds *DataSource) BuildKeyInfo(selfSchema *expression.Schema, childSchema []
 		if path.IsIntHandlePath {
 			continue
 		}
-		if newKey := checkIndexCanBeKey(path.Index, ds.Columns, selfSchema); newKey != nil {
+		if _, newKey := checkIndexCanBeKey(path.Index, ds.Columns, selfSchema); newKey != nil {
 			selfSchema.Keys = append(selfSchema.Keys, newKey)
 		}
+		if uniqueKey, _ := checkIndexCanBeKey(path.Index, ds.Columns, selfSchema); uniqueKey != nil {
```

Comment:
Please avoid this redundant `checkIndexCanBeKey` call.

Reply:
done